cmpserver/plugin/plugin.go

```diff
@@ -128,7 +129,26 @@ func environ(envVars []*apiclient.EnvEntry) []string {
 }
 
 // GenerateManifest runs generate command from plugin config file and returns generated manifest files
-func (s *Service) GenerateManifest(ctx context.Context, q *apiclient.ManifestRequest) (*apiclient.ManifestResponse, error) {
+func (s *Service) GenerateManifest(stream apiclient.ConfigManagementPluginService_GenerateManifestServer) error {
+	workDir, env, err := cmp.ReceiveApplicationStream(stream.Context(), stream)
+	if err != nil {
+		return fmt.Errorf("generate manifest error receiving stream: %s", err)
+	}
+	defer os.RemoveAll(workDir)
```

Comment:
Agree with 1 but disagree with 2. The workDir path is never sent back to the repo-server and even if it was the case that path doesn't exist there as it is a separate container.
Added warning log in case of error.

Reply:
> workDir path is never sent back to the repo-server 

I don't think we know that for sure. CMP server is designed to run arbitrary code. We don't know the vulnerabilities of that code. Writing the current working directory as part of a manifest might be one of its vulnerabilities.

> even if it was the case that path doesn't exist there as it is a separate container

I don't think that matters. An attacker who can affect manifest generation (via source files, App config, parameters, etc.) could use info about the CMP to attempt a path traversal on the CMP.

> Added warning log in case of error.

I'm not sure warning is strong enough. Our only justification for not sanitizing UUID paths from errors (or from manifests) is that we are _certain_ that the path will not exist by the time repo-server receives the message. If we are not meeting that guarantee then we should either 1) sanitize messages so that persistent paths aren't leaked or 2) be very loud about the fact that paths which should be transient are not in fact being transient.